pkg/k8shandler/collection.go

```diff
@@ -224,6 +386,82 @@ func getFluentdPodSpec(logging *logging.ClusterLogging, elasticsearchAppName str
+		{Name: "BUFFER_QUEUE_LIMIT", Value: "32"},
+		{Name: "BUFFER_SIZE_LIMIT", Value: "8m"},
+		{Name: "FILE_BUFFER_LIMIT", Value: "256Mi"},
+		{Name: "RSYSLOG_CPU_LIMIT", ValueFrom: &v1.EnvVarSource{ResourceFieldRef: &v1.ResourceFieldSelector{ContainerName: "rsyslog", Resource: "limits.cpu"}}},
+		{Name: "RSYSLOG_MEMORY_LIMIT", ValueFrom: &v1.EnvVarSource{ResourceFieldRef: &v1.ResourceFieldSelector{ContainerName: "rsyslog", Resource: "limits.memory"}}},
+	}
+
+	rsyslogContainer.VolumeMounts = []v1.VolumeMount{
+		{Name: "runlogjournal", MountPath: "/run/log/journal"},
+		{Name: "varlog", ReadOnly: true, MountPath: "/var/log"},
+		{Name: "varlibdockercontainers", ReadOnly: true, MountPath: "/var/lib/docker"},
+		{Name: "bin", ReadOnly: true, MountPath: "/opt/app-root/bin"},
+		{Name: "main", ReadOnly: true, MountPath: "/etc/rsyslog/conf"},
+		{Name: "config", ReadOnly: true, MountPath: "/etc/rsyslog.d"},
+		{Name: "certs", ReadOnly: true, MountPath: "/etc/rsyslog/keys"},
+		{Name: "dockerhostname", ReadOnly: true, MountPath: "/etc/docker-hostname"},
+		{Name: "localtime", ReadOnly: true, MountPath: "/etc/localtime"},
+		{Name: "filebufferstorage", MountPath: "/var/lib/rsyslog"},
+	}
+
+	rsyslogContainer.SecurityContext = &v1.SecurityContext{
+		Privileged: utils.GetBool(true),
+	}
+
+	rsyslogContainer.Command = []string{
+		"/bin/sh",
+	}
+
+	rsyslogContainer.Args = []string{
+		"/opt/app-root/bin/rsyslog.sh",
+	}
+
+	rsyslogPodSpec := utils.PodSpec(
+		"rsyslog",
+		[]v1.Container{rsyslogContainer},
+		[]v1.Volume{
+			{Name: "runlogjournal", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/run/log/journal"}}},
+			{Name: "varlog", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/var/log"}}},
+			{Name: "varlibdockercontainers", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/var/lib/docker"}}},
+			{Name: "bin", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{LocalObjectReference: v1.LocalObjectReference{Name: "rsyslog-bin"}}}},
+			{Name: "main", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{LocalObjectReference: v1.LocalObjectReference{Name: "rsyslog-main"}}}},
+			{Name: "config", VolumeSource: v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{LocalObjectReference: v1.LocalObjectReference{Name: "rsyslog"}}}},
+			{Name: "certs", VolumeSource: v1.VolumeSource{Secret: &v1.SecretVolumeSource{SecretName: "rsyslog"}}},
+			{Name: "dockerhostname", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/etc/hostname"}}},
+			{Name: "localtime", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/etc/localtime"}}},
+			{Name: "filebufferstorage", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/var/lib/rsyslog"}}},
+		},
+	)
+
+	rsyslogPodSpec.PriorityClassName = "cluster-logging"
+
+	rsyslogPodSpec.NodeSelector = map[string]string{
```

Comment:
I think I made this change in #23 but we likely want this to come from the CR instead of being hardcoded

Reply:
Ok.  Likely #23 will merge first and I will rebase this on top of it.